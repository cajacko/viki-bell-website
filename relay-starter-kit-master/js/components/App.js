import React from 'react';
import Relay from 'react-relay';

class App extends React.Component {
  render() {
    if (!this.props.posts) {
        return false;
    }

    return (
      <div>
        <h1>Widget list</h1>
        <ul>
          {this.props.posts.edges.map(edge =>
            <li key={edge.node.id}>{edge.node.title} (ID: {edge.node.id})</li>
          )}
        </ul>
      </div>
    );
  }
}

export default Relay.createContainer(App, {
  fragments: {
    posts: () => Relay.QL`
      fragment on Query {
        posts(first: 10) {
          edges {
            node {
              id,
              title,
            },
          },
        },
      }
    `,
  },
});
