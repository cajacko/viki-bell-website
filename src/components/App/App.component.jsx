import { Provider } from 'react-redux';
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Route, Switch } from 'react-router-dom';
import HomeView from 'components/HomeView/HomeView';
import ScrollToTop from 'components/ScrollToTop/ScrollToTop';
import removeLoading from 'helpers/removeLoading';
import PostsView from 'components/PostsView/PostsView';
import PostView from 'components/PostView/PostView';
import FourOhFourView from 'components/FourOhFourView/FourOhFourView';
import SitemapView from 'components/SitemapView/SitemapView';
import CategoriesView from 'components/CategoriesView/CategoriesView';
import CategoryView from 'components/CategoryView/CategoryView';
import PageView from 'components/PageView/PageView';
import AppError from 'components/AppError/AppError';

class App extends Component {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    this.props.getGlobalContent();
  }

  componentWillReceiveProps(props) {
    if (this.props.status === 'loading' && props.status !== 'loading') {
      removeLoading();
    }
  }

  render() {
    if (this.props.status === 'loading') {
      return null;
    }

    if (this.props.status === 'error') {
      return <AppError />;
    }

    const Router = this.props.Router;

    return (
      <Provider store={this.props.store}>
        <Router location={this.props.location} context={this.props.context}>
          <ScrollToTop>
            <Switch>
              <Route path="/" exact component={HomeView} />
              <Route path="/sitemap" exact component={SitemapView} />
              <Route path="/posts" exact component={PostsView} />
              <Route path="/posts/:slug" component={PostView} />
              <Route path="/categories" exact component={CategoriesView} />
              <Route path="/categories/:slug" component={CategoryView} />
              <Route path="/:slug" component={PageView} />
              <Route component={FourOhFourView} />
            </Switch>
          </ScrollToTop>
        </Router>
      </Provider>
    );
  }
}

App.propTypes = {
  location: PropTypes.string,
  // eslint-disable-next-line
  context: PropTypes.object,
};

App.defaultProps = {
  location: undefined,
  context: undefined,
};

export default App;
