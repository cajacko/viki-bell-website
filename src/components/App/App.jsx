import { Provider } from 'react-redux';
import React from 'react';
import PropTypes from 'prop-types';
import { Route, Switch } from 'react-router-dom';
import HomeView from 'components/HomeView/HomeView';
import ScrollToTop from 'components/ScrollToTop/ScrollToTop';
import removeLoading from 'helpers/removeLoading';
import PostsView from 'components/PostsView/PostsView';
import PostView from 'components/PostView/PostView';
import FourOhFourView from 'components/FourOhFourView/FourOhFourView';
import SitemapView from 'components/SitemapView/SitemapView';
import CategoriesView from 'components/CategoriesView/CategoriesView';
import CategoryView from 'components/CategoryView/CategoryView';
import PageView from 'components/PageView/PageView';

removeLoading();

const App = ({ Router, store, location, context }) => (
  <Provider store={store}>
    <Router location={location} context={context}>
      <ScrollToTop>
        <Switch>
          <Route path="/" exact component={HomeView} />
          <Route path="/sitemap" exact component={SitemapView} />
          <Route path="/posts" exact component={PostsView} />
          <Route path="/posts/:slug" component={PostView} />
          <Route path="/categories" exact component={CategoriesView} />
          <Route path="/categories/:slug" component={CategoryView} />
          <Route path="/:slug" component={PageView} />
          <Route component={FourOhFourView} />
        </Switch>
      </ScrollToTop>
    </Router>
  </Provider>
);

App.propTypes = {
  location: PropTypes.string,
  // eslint-disable-next-line
  context: PropTypes.object,
};

App.defaultProps = {
  location: undefined,
  context: undefined,
};

export default App;
