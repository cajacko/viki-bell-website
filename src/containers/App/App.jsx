import { Provider } from 'react-redux';
import React from 'react';
import PropTypes from 'prop-types';
import { Route } from 'react-router-dom';
import HomeView from 'components/HomeView/HomeView';
import ScrollToTop from 'containers/ScrollToTop/ScrollToTop';
import removeLoading from 'helpers/removeLoading';

removeLoading();

// eslint-disable-next-line
const App = ({ Router, store, location, context }) => (
  <Provider store={store}>
    <Router location={location} context={context}>
      <ScrollToTop>
        <Route path="/" component={HomeView} />
      </ScrollToTop>
    </Router>
  </Provider>
);

// <Route path="/posts" component={PostsView} />
// <Route path="/categories" component={CategoriesView} />
// <Route path="/posts/:slug" component={PostView} />
// <Route path="/categories/:slug" component={CategoryView} />
// <Route path="/sitemap" component={SitemapView} />
// <Route path="/:slug" component={PageView} />
// <Route path="*" component={FourOhFourView} />

App.propTypes = {
  location: PropTypes.string,
  // eslint-disable-next-line
  context: PropTypes.object,
};

App.defaultProps = {
  location: undefined,
  context: undefined,
};

export default App;
