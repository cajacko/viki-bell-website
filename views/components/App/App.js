import React from 'react';
import PostLoopContainer from 'containers/PostLoop/PostLoop';
import SiteNav from 'components/SiteNav/SiteNav';
import SiteHeader from 'components/SiteHeader/SiteHeader';
import About from 'components/About/About';

class App extends React.Component {
  componentDidMount() {
    window.initialRender = true;
    const now = Math.floor(Date.now());
    const loading = document.getElementById('loading');

    if (window.noLoading) {
      loading.remove();
    } else {
      const loadingTime = now - window.startTime;

      // Otherwise looks like flash of content
      const minimumLoadTime = 1000;
      let additionalLoadTime = minimumLoadTime - loadingTime;

      if (additionalLoadTime < 0) {
        additionalLoadTime = 0;
      }

      setTimeout(() => {
        loading.classList.remove('loading');

        loading.addEventListener('transitionend', () => {
          loading.remove();
        }, true);
      }, additionalLoadTime);
    }
  }

  render() {
    return (
      <div>
        <SiteNav />
        <SiteHeader />
        <PostLoopContainer query="/" />
        <About />
      </div>
    );
  }
}

export default App;
