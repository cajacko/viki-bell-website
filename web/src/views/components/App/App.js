/* @flow */

import React from 'react';
import Posts from 'containers/PostLoop/PostLoop';

const App = () => (
  <div>
    <Posts />
    <Posts inverseColours recommendedPosts />
  </div>
);

export default App;
