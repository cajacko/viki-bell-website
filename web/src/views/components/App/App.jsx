/* @flow */

import React from 'react';
import Posts from 'components/Posts/Posts';

class App extends React.Component {
  constructor() {
    super();

    this.state = {
      posts: [
        {
          id: 1,
          title: 'Post Title 1',
          excerpt: 'Post excerpt of some kind',
        },
        {
          id: 2,
          title: 'Post Title 2',
          excerpt: 'I am an excerpt',
        },
      ],
    };
  }

  state: { posts: Array<Object> };

  render() {
    return (
      <div>
        <h1>I am App from View</h1>
        <Posts
          posts={this.state.posts}
        />
      </div>
    );
  }
}

export default App;
