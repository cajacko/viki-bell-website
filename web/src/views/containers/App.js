import React, { Component } from 'react';

import {
  QueryRenderer,
  graphql,
} from 'react-relay';

import environment from '../relayEnvironment';
import PostLoop from './PostLoop';
// import Post from './Post';

class App extends Component {
  render() {
    return (
      <div className="App">
        <h2>App</h2>
        <QueryRenderer
          environment={environment}

          query={graphql`
            query AppQuery {
              ...PostLoop_data
            }
          `}

          variables={{}}

          render={({ error, props }) => {
            if (error) {
              return <div>{error.message}</div>;
            } else if (props) {
              // eslint-disable-next-line
              console.warn('App', props);

              return <PostLoop data={props} />;
            }
            return <div>Loading</div>;
          }}
        />
      </div>
    );
  }
}

export default App;
