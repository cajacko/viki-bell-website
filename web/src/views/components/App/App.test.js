import React from 'react';
import App from 'components/App/App';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <App />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
