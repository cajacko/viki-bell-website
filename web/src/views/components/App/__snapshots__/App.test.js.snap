// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div>
    Loading
  </div>
  <h1>
    I am App from View
  </h1>
</div>
`;
